fenix/app/src/test/java/org/mozilla/fenix/share/SaveToPDFMiddlewareTest.kt

```diff
@@ -39,190 +48,233 @@ class SaveToPDFMiddlewareTest {
     // Only ERROR_PRINT_SETTINGS_SERVICE_NOT_AVAILABLE is available for testing
     class MockGeckoPrintException() : GeckoSession.GeckoPrintException()
 
+    @Before
+    fun setup() {
+        appStore = mockk(relaxed = true)
+        every { testContext.components.appStore } returns appStore
+    }
+
     @Test
-    fun `GIVEN a save to pdf request WHEN it fails unexpectedly THEN unknown failure telemetry is sent`() = runTestOnMain {
-        val exceptionToThrow = RuntimeException("reader save to pdf failed")
-        val middleware = SaveToPDFMiddleware(testContext)
-        val mockEngineSession: EngineSession = mockk<EngineSession>().apply {
-            every {
-                checkForPdfViewer(any(), any())
-            } answers {
-                secondArg<(Throwable) -> Unit>().invoke(exceptionToThrow)
+    fun `GIVEN a save to pdf request WHEN it fails unexpectedly THEN unknown failure telemetry is sent`() =
```

Comment:
nit: 
```suggestion
    fun `GIVEN a save to pdf request WHEN it fails unexpectedly THEN unknown failure telemetry is sent AND a snackbar error is shown`() =
```
(A few other tests could be renamed too.)

Reply:
Done